monitor/services/src/main/java/org/datacleaner/monitor/server/controllers/ResultsFolderController.java

```diff
@@ -49,16 +51,52 @@
     @RolesAllowed(SecurityRoles.VIEWER)
     @RequestMapping(method = RequestMethod.GET, produces = "application/json")
     @ResponseBody
-    public List<Map<String, String>> resultsFolderJson(@PathVariable("tenant") final String tenant) {
+    public List<Map<String, String>> resultsFolderJson(@PathVariable("tenant") final String tenant,
+            @RequestParam(value = "time", required = false) final String timestamp) {
+        
+        
+        if (timestamp == null || Long.valueOf(timestamp) < 0) {
+            return resultsFolderJsonHelper(tenant);
+        }
+
+        final TenantContext context = _tenantContextFactory.getContext(tenant);
+        final Timestamp searchedTimestamp = new Timestamp(Long.valueOf(timestamp));
+        final RepositoryFolder resultsFolder = context.getResultFolder();
+
+        final List<Map<String, String>> result = new ArrayList<>();
+
+        {
+            final List<RepositoryFile> files = resultsFolder.getFiles(null, 
+                    FileFilters.ANALYSIS_RESULT_SER.getExtension());
+            for (final RepositoryFile file : files) {
+                final Map<String, String> map = new HashMap<>();
+                final String name = file.getName();
+                // get the timestamp of the job
+                final String timestampString = name.substring(name.lastIndexOf("-") + 1, name.indexOf("."));
+                final Timestamp jobTimestamp = new Timestamp(Long.valueOf(timestampString));
+                if (jobTimestamp.after(searchedTimestamp)) {
+                    map.put("filename", name);
+                    map.put("repository_path", file.getQualifiedPath());
+                }
+                if (map.size() > 0) {
```

Comment:
Errr, isn't map size only > 0 if the the previous `if` is true? So why not move it into the previous `if`?